baseapp/abci.go

```diff
@@ -969,6 +978,8 @@ func SplitABCIQueryPath(requestPath string) (path []string) {
 func (app *BaseApp) getContextForProposal(ctx sdk.Context, height int64) sdk.Context {
 	if height == 1 {
 		ctx, _ = app.deliverState.ctx.CacheContext()
+		// TODO: clear all context data set during InitChain to avoid inconsistent behavior
+		ctx = ctx.WithBlockHeader(cmtproto.Header{}).WithChainID("")
```

Comment:
I don't understand this part. Why do we need to set an empty chain-id?

Reply:
I'm now adding the chainID to baseapp through an option, not sure if it's the cleanest way tho

Reply:
Should we check if the app's chain id is equal to the req.ChainID in every step? Like BeginBlock, EndBlock, etc.

Reply:
We could, yes.

Reply:
I think that's what I did, whenever you have time give it a quick read just to know if I'm on the right track. I'll add the checks I mentioned above in the meantime